react-ui/src/pages/submission/SubmissionForm.jsx

```diff
@@ -357,33 +356,45 @@ function SubmissionForm({
     }
   };
 
-  const disableSubmitting = submitting || isMutating;
+  const nameCharRemaining = CHAR_LIMIT_NAME - formState.name.value.length;
+  const descriptionCharRemaining = CHAR_LIMIT_DESCRIPTION - formState.description.value.length;
+  const disableEditing = submitting || isMutating;
+  const disableSubmitting = disableEditing || nameCharRemaining < 0 || descriptionCharRemaining < 0;
 
   const classes = useStyles();
 
   return (
-    <ProtectedRoute
-      message="You must log in with Reddit to submit a flag"
-      showCancel={false}
-    >
+    <ProtectedRoute message="You must log in with Reddit to submit a flag" showCancel={false}>
       <form id="submission-form">
-        <Fieldset disabled={disableSubmitting || submissionExpired}>
-          <TextField
-            id="username"
-            variant="filled"
-            label="Username"
-            disabled
-            value={username}
-          />
+        <Fieldset disabled={disableEditing || submissionExpired}>
+          <TextField id="username" variant="filled" label="Username" disabled value={username} />
           <TextField
             id="name"
             name="name"
             color="secondary"
             variant="filled"
-            helperText={formState.name.error || 'A concise name for your flag'}
+            helperText={(
+              <div>
```

Comment:
This text ends up being quite a lot larger font than the helper text on other fields. Is there a reason these divs are needed? Could just use one text string, or use `<span>` elements instead of `<Typography>`. I'm also not sure how easy it is to read in one long sentence, perhaps they should be over two lines, or my preference would be to have the new counter right-aligned with the rest of the text left-aligned in a flex-box.

Reply:
Happy to change it, just let me know what you prefer!

Reply:
Let's do the flex-box with `count/max` right aligned, and the helper text / error message left aligned taking up as much space as possible, please.